Cargo.toml

```diff
@@ -25,6 +25,16 @@ publish = false
 
 [features]
 default = []
```

Comment:
Do we want to include all features by default and make libraries set `default-features = false`?

Reply:
@carllerche preferred the default being the API shim, but it was a brief question in the Discord channel. Not sure if he has any more thoughts on it now.

Reply:
For new users it might be a bit annoying to start using Mio, as well as for all people that come from v0.6. I see Tokio also doesn't include anything by default, I'll leave it up to @carllerche then. I don't really have a strong opinion either way.

Reply:
Created #1216.